get_http.go

```diff
@@ -118,12 +120,60 @@ func (g *HttpGetter) GetFile(dst string, u *url.URL) error {
 		g.Client = httpClient
 	}
 
-	resp, err := g.Client.Get(u.String())
+	// check to see whether user has specified a range of bytes to download.
+	// if user has, but the range is invalid, fall back to downloading the
+	// whole file
+	byteRange, rangeErr := getByteRange(u)
+	if rangeErr != nil {
+		// log that we are downloading whole file even though user
+		// requested a byte range.
+		log.Printf("%s; going to disregard range request and download entire file",
+			rangeErr)
+	}
+
+	// First request here (preReq), is going to be used to determine whether we
+	// can make a ranged request. Second one, (req) is used for the actual
+	// download
+	preReq, err := http.NewRequest("HEAD", u.String(), nil)
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequest("HEAD", u.String(), nil)
```

Comment:
I think you can just set `"GET"` all the way up here now that you have two separate HTTP calls, and remove the line below that sets this!